bln-mode.el

```diff
@@ -66,32 +66,33 @@
 
 ;;; Code:
 
-(defvar bln-beg -1)
-(defvar bln-end -1)
+(defvar bln-beg-end '(-1 . -1))
 (defvar bln-prev-mid -1)
 
 ;;;###autoload
 (defun bln-backward-half ()
   "This function is used in combination with `bln-forward-half' to provide binary line navigation (see `bln-mode')."
   (interactive)
-  (if (/= bln-prev-mid (point))
-      (setq bln-beg -1 bln-end -1)
-    (setq bln-end bln-prev-mid))
-  (if (< bln-beg 0) (setq bln-beg (line-beginning-position)
-			  bln-end (point)))
-  (setq bln-prev-mid (/ (+ bln-beg bln-end) 2))
+  (setq bln-beg-end
+        (if (/= bln-prev-mid (point))
+            '(-1 . -1)
+          `(,(car bln-beg-end) . ,bln-prev-mid)))
```

Comment:
I prefer the form (cons expr1 expr2) to `(,expr1 . ,expr2) because I think it is easier to read by novices. If you agree, would you mind changing it throughout the file?

BTW. I'm not experienced in handling pull requests either, but I don't think it is necessary to create another pull request after you make further commits. Any new commits on the branch you asked me to merge will be reflected in the PR as well, I believe.